### ISSUE TITLE
Performance optimizations needed for Annotation companion window when their are many annotations

### ISSUE THREAD

**Issue by mejackreed** [ProjectMirador/mirador#2915]
Using manifest https://iiif.europeana.eu/presentation/9200301/BibliographicResource_3000126341277/manifest after #2914 is merged. This newspaper page has 2292 annotations. The annotations panel is causing a page slowdown when rendering.

**Comment by alansutherland** [comment]
any updates on this issue @mejackreed ?

**Comment by mejackreed** [comment]
There were some performance optimizations in https://github.com/ProjectMirador/mirador/pull/2920 for manifest selections. I'm not sure of anyone who has looked at this issue yet. However I think a few approaches could help:

 - look to see if there is an expensive select happening many times, if so memoize it using re-reselect (similar to #2920)
 - look at windowing the annotation list display using react-window
 - see if there are design considerations that could help (with 1000's of annotations on a canvas, should/could we implement pagination?)

**Comment by mejackreed** [comment]
And to be clear, we would be open to a pull request from anyone that aimed to resolve these performance issues.

**Comment by alansutherland** [comment]
thanks for the advice will look into using reselect / react-window
we're loading over 2000 annotations which is no doubt slowing it down.
If we have any success developing a plugin for this we'll submit a pull request

**Comment by alansutherland** [comment]
@mejackreed I have another issue https://github.com/ProjectMirador/mirador/issues/2971 
any chance you could help?

**Comment by marlo-longley** [comment]
This is the same fix pattern as #2856 (canvas index virtualization with react-window). Probably should be looked at by the same developer
